core/include/usBundleResource.h

```diff
@@ -28,13 +28,14 @@
 
 #include <ostream>
 #include <vector>
+#include <memory>
 
 US_MSVC_PUSH_DISABLE_WARNING(4099) // type name first seen using 'struct' now seen using 'class'
```

Comment:
This warning disabling push/pop can be removed by changing "friend class ::std::hash<BundleResource>" to "friend struct ::std::hash<BundleResource>".

std::hash is a struct .... http://en.cppreference.com/w/cpp/utility/hash